UnicornRender/video/source/vulkan/Renderer.cpp

```diff
@@ -1503,9 +1493,9 @@ bool Renderer::AllocateMaterial(const Mesh& mesh, VkMesh& vkmesh)
 
             auto result = m_vkLogicalDevice.allocateDescriptorSets(&allocInfo, &descriptorSet);
 
-            if(result != vk::Result::eSuccess)
+            if(result != vk::Result::eSuccess || !vkTexture->IsInitialized())
             {
-                LOG_ERROR("Can't allocate descriptor sets!");
+                LOG_ERROR("Can't allocate sampler descriptor sets!");
 
                 vkTexture->Delete();
```

Comment:
Wait didn't you rename it to `FreeData()`? What's going on here? Why is this called if `vkTexture` wasn't initialized? What's up with all these names that seem to do the same thing

Reply:
unicorn::Texture have freeData, and here we have Delete.